File: code_examples/4-Functions/7-Recursion/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program calculates factorial
*/

using System;

/*
 * The Program class
*/
class Program {
        // recursive function calculating factorial
        static int Factorial(int number) {
            if (number == 1) {
                return 1;
            }
            else {
                return number * Factorial(number - 1);
            }
        }

    static void Main() {
        // this function gets user input
        int userInputInt;

        // input
        Console.Write("Enter a positive integer: ");
        userInputInt = int.Parse(Console.ReadLine());

        if (userInputInt < 0) {
            Console.WriteLine("That was not a positive integer.");
        } else {
            // call function
            int factorialAnswer = Factorial(userInputInt);
            Console.WriteLine($"The factorial of {userInputInt} is {factorialAnswer}");
        }

        Console.WriteLine("\nDone.");
    }
}

File: code_examples/4-Functions/7-Recursion/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program calculates factorial
 */

package main

import "fmt"

func factorial(number int) int {
	// recursive function calculating factorial
	if number == 1 {
		return 1
	} else {
		return number * factorial(number-1)
	}
}

func main() {
	// this function gets user input
	var userInputInt int

	// input
	fmt.Print("Enter a positive integer: ")
	fmt.Scanln(&userInputInt)

	if userInputInt < 0 {
		fmt.Println("That was not a positive integer.")
	} else {
		// call function
		factorialAnswer := factorial(userInputInt)
		fmt.Printf("The factorial of %d is %d\n", userInputInt, factorialAnswer)
	}

	fmt.Println("\nDone.")
}
